biz.aQute.bndlib/src/aQute/bnd/apiguardian/APIGuardianAnnotations.java

```diff
@@ -0,0 +1,67 @@
+package aQute.bnd.apiguardian;
+
+import java.util.Map.Entry;
+
+import aQute.bnd.header.Attrs;
+import aQute.bnd.header.OSGiHeader;
+import aQute.bnd.header.Parameters;
+import aQute.bnd.osgi.Analyzer;
+import aQute.bnd.osgi.Clazz;
+import aQute.bnd.osgi.Constants;
+import aQute.bnd.osgi.Instruction;
+import aQute.bnd.osgi.Instructions;
+import aQute.bnd.service.AnalyzerPlugin;
+
+public class APIGuardianAnnotations implements AnalyzerPlugin {
+
+	private final static String API_ANNOTATION = "org/apiguardian/api/API";
+	private final static String	INTERNAL_STATUS	= "INTERNAL";
+	private final static String	STATUS_PROPERTY	= "status";
+
+	@Override
+	public boolean analyzeJar(Analyzer analyzer) throws Exception {
+		// Opt-in is required.
+		Parameters header = OSGiHeader.parseHeader(analyzer.getProperty(Constants.EXPORT_APIGUARDIAN));
+		if (header.isEmpty())
+			return false;
+
+		Parameters exportPachages = analyzer.getExportPackage();
+
+		Instructions instructions = new Instructions(header);
+
+		Parameters apiGuardianPackages = new Parameters();
+
+		for (Clazz c : analyzer.getClassspace()
+			.values()) {
+
+			if (c.isModule() || c.isInnerClass() || c.isSynthetic()) {
+				continue;
+			}
+
+			for (Entry<Instruction, Attrs> entry : instructions.entrySet()) {
+				Instruction instruction = entry.getKey();
+				if (instruction.matches(c.getFQN())) {
+					if (instruction.isNegated()) {
+						break;
+					}
+
+					if (c.annotations(API_ANNOTATION)
+						.map(ann -> (String) ann.get(STATUS_PROPERTY))
+						.anyMatch(status -> !INTERNAL_STATUS.equals(status))) {
+
+						apiGuardianPackages.add(c.getClassName()
+							.getPackageRef()
+							.getFQN(), entry.getValue());
+					}
+				}
+			}
+		}
+
+		exportPachages.putAll(apiGuardianPackages);
```

Comment:
You should probably use `mergeWith` instead of `putAll` in case there are already attrs from an existing `-exportcontents`.

Reply:
I still think you need to merge here.